src/util/util.ts

```diff
@@ -66,10 +66,19 @@ if (typeof Element !== 'undefined' && !Element.prototype.closest) {
   };
 }
 
-export function closest(element: HTMLElement, selector): HTMLElement | null {
+export function closest(element: HTMLElement, selector?: string): HTMLElement | null {
   if (!selector) {
     return null;
   }
 
+  /*
```

Comment:
I think we should do thing a little bit differently here because we can't just access `document` like that.

Any angular app could rendered server-side, where basically `document` won't be defined. So in order to properly use it, we usually have to get it by injection so that DI can provide a _special_ one in environment where there is no `document`. (Have a [look](https://github.com/ng-bootstrap/ng-bootstrap/search?q=%40Inject%28DOCUMENT&unscoped_q=%40Inject%28DOCUMENT) at how we do it)

But here, as we are in an utility function, it would mean we need to pass it as an argument, to have access to DI from the caller standpoint.

I found this a bit heavy, for such a small use case. (It apparently only fails in EdgeHTML)

What about we simply
```typescript
try {
  return element.closest(selector)
} catch(err) { 
  return null;
}
```

WDYT ?

Reply:
Nice catch. It slipped my mind. 😛 

I'm a bit hesitant to use try-catch since any error would be caught. How about using a more exact check? 

```typescript
if (typeof element.closest === 'undefined') {
  return null;
}
```

Beware though; in modern browsers `document.documentElement.closest('html')` would return `document`. By returning null always, we get a differing behaviour in Edge. How would you go about that problem?

Reply:
Ping @benouat. Would you be ok with that behaviour and check? :)

Reply:
Yes, go ahead with that!

Don't forget to comment out things (this is needed for old edgeHTML generation) because I am pretty sure in 3 months from now we'll have forgotten why we do have this test!

Reply:
@benouat Cool. I've changed the commit and code according to discussion. :)